pkg/router/routes_test.go

```diff
@@ -310,247 +317,165 @@ func makeFake(t *testing.T, r *endpoints.Resource, resourceType, namespace, reso
-		if err != nil {
-			t.Fatalf("Error deleting service '%s': %v\n", svc.Name, err)
-		}
-		_, err = r.K8sClient.CoreV1().Services(svc.Namespace).Create(&svc)
-		if err != nil {
-			t.Fatalf("Error recreating service '%s': %v\n", svc.Name, err)
-		}
-	}
-
-	timeout := time.After(10 * time.Second)
-	var extensionCreates int
-	var extensionDeletes int
-	subChan := subscriber.SubChan()
-	// Wait until all extension creates/deletes are registered by extensionInformer
-	t.Logf("waiting for registration")
-	t.Logf("services:" + string(len(extensionServices)))
-	t.Logf("start time: " + time.Now().String())
-	t.Logf("--- ---")
-	counter := 0
-	for {
-		counter++
-		select {
-		case <-timeout:
-			t.Fatal("Timed out waiting for expected services to be registered")
-			t.Logf("fatal time: " + time.Now().String())
-			t.Logf("fatal counter: " + strconv.Itoa(counter))
-			t.Logf("extensionCreates:")
-			t.Logf(strconv.Itoa(extensionCreates))
-			t.Logf("extensionDeletes:")
-			t.Logf(strconv.Itoa(extensionDeletes))
-		case event := <-subChan:
-			t.Logf("counter: " + strconv.Itoa(counter))
-			t.Logf("time: " + time.Now().String())
-			switch event.MessageType {
-			case broadcaster.ExtensionCreated:
-				extensionCreates++
-				t.Logf("extensionCreates:")
-				t.Logf(strconv.Itoa(extensionCreates))
-			case broadcaster.ExtensionDeleted:
-				extensionDeletes++
-				t.Logf("extensionDeletes:")
-				t.Logf(strconv.Itoa(extensionDeletes))
 			}
-			t.Logf("")
-		}
-		// All events captured from Create->Delete->Create above
-		if extensionCreates == 2*len(extensionServices) && extensionDeletes == len(extensionServices) {
-			t.Logf("--- events captured ---")
-			break
-		}
-	}
-	t.Logf("final time: " + time.Now().String())
-	t.Logf("final counter: " + strconv.Itoa(counter))
-	t.Logf("extensionCreates:")
-	t.Logf(strconv.Itoa(extensionCreates))
-	t.Logf("extensionDeletes:")
-	t.Logf(strconv.Itoa(extensionDeletes))
+			// Create extension service and await corresponding event
+			_, err := r.K8sClient.CoreV1().Services(installNamespace).Create(extensionService)
+			if err != nil {
+				t.Fatalf("Error creating service '%s': %v\n", extensionService.Name, err)
+			}
+			select {
+			case event := <-subscriber.SubChan():
+				switch event.MessageType {
+				case broadcaster.ExtensionCreated:
+					break
+				default:
+					t.Fatalf("Unexpected event message type: %s", event.MessageType)
+				}
+			case <-time.After(time.Second):
+				t.Fatalf("Timed out waiting for extension creation")
+			}
 
-	// Labels not supported on fake client, manual filter
-	serviceList, err := r.K8sClient.CoreV1().Services(installNamespace).List(metav1.ListOptions{})
-	if err != nil {
-		t.Fatalf("Error obtaining services from K8sClient")
-	}
+			// Assert routes
+			replyCode := http.StatusNoContent
+			extensionProxyRoot := fmt.Sprintf("%s%s/%s", server.URL, ExtensionRoot, extensionService.Name)
+			defer gock.Off()
```

Comment:
What does this stuff do? `Assert routes` is pretty vague, we've got it now as an import but I shouldn't have to go look it up to understand this block and why it's now necessary

Reply:
Before there was a dummy extension that the test would explicitly make requests to, which "mocked" an actual extension. Using gock, the requests actually go through the dashboard test server and the proxy requests are intercepted (the transport level is intercepted on clients, where both the dashboard and gock are using the default client), which is a more appropriate mock/behavior test.

Reply:
I'll update the comment

Reply:
Sounds much better - excellent, thanks